From: Jiri Kosina <jkosina@suse.cz>
Subject: HID: fix kabi breakage
Patch-mainline: never

Commit 1f5b75fd69b4b1185ae3b7be4cd544c5f503d888 added usage_index to
struct hid_usage. struct hid_usage is not embedded in any other structure
directly, so it's safe to avoid the kABI brekage by GENKSYMS ifdef.

Signed-off-by: Jiri Kosina <jkosina@suse.cz>

Index: current/include/linux/hid.h
===================================================================
--- current.orig/include/linux/hid.h	2014-10-29 23:49:17.474543134 +0100
+++ current/include/linux/hid.h	2014-10-29 23:49:17.466543134 +0100
@@ -342,13 +342,15 @@ struct hid_collection {
 struct hid_usage {
 	unsigned  hid;			/* hid usage code */
 	unsigned  collection_index;	/* index into collection array */
-	unsigned  usage_index;		/* index into usage array */
 	/* hidinput data */
 	__u16     code;			/* input driver code */
 	__u8      type;			/* input driver type */
 	__s8	  hat_min;		/* hat switch fun */
 	__s8	  hat_max;		/* ditto */
 	__s8	  hat_dir;		/* ditto */
+#ifndef __GENKSYMS__
+	unsigned  usage_index;		/* index into usage array */
+#endif
 };
 
 struct hid_input;
